src/features/channels/components/channel-list/ChannelList.tsx

```diff
@@ -1,31 +1,55 @@
 import { useEffect, useState } from 'react';
 import { useAppDispatch, useAppSelector } from '@metis/store/hooks';
+import { Channel } from '@metis/features/channels/types/channel';
 
 import Spinner from '@metis/common/components/ui/spinner/Spinner';
 import ChannelListItem from './channel-list-item/ChannelListItem';
 
 import { selectState } from '../../store/channel.slice';
-import { findChannels } from '../../store/channel.actions';
+import { findChannels, getHiddenChannels, hideChannel } from '../../store/channel.actions';
 
 const ChannelList = () => {
+  const channelInitialState: Channel[] = [];
   const dispatch = useAppDispatch();
   const { channels, isLoading } = useAppSelector(selectState);
   const [selected, setSelected] = useState('');
+  const [currentChannels, setChannels] = useState(channelInitialState);
+
+  const getChannelList = () => {
```

Comment:
We should leverage the business logic to the reducers. What I'd suggest to do, is to dispatch the getHiddenChannels action, create a hiddenChannels state variable in the channels slice, filter out the list, and use a selector to pull that state variable, this way, we would avoid too much login in the component and keep it clean, and we could potentially reuse that selector to other components as well.